common/trace/src/trace.rs

```diff
@@ -452,7 +452,9 @@ pub fn is_selected(node: (&'static str, u64)) -> bool {
         match &SAMPLING_CONFIG {
             Some(Sampling(sampling)) => {
                 if let Some(sampling_rate) = sampling.get(node.0) {
-                    node.1 % sampling_rate.denominator < sampling_rate.nominator
+                    node.1.checked_rem(sampling_rate.denominator).expect(
+                        "Sampling rate denominator has a value of 0, resulting in integer overflow",
+                    ) < sampling_rate.nominator
```

Comment:
I don't think we should have panics in a crate that is not critical to the flow

Reply:
@mimoo ok instead of panicking I return true now if denominator is 0 . I kept the wrapping_rem so that clippy doesn't complain but this should do the trick now